### PR TITLE
Add more trait re-exports to prelude module

### PR DESCRIPTION
**Pull request by icedrocket**

### CODE REVIEW

**Comment by jessebraham** [comment]
Thanks for the PR.

What was the motivation behind these changes? For example `GpioExt` wasn't exposed because it _shouldn't_ be exposed, the `split()` function is called in `IO::new()`; is there an issue with  this?

**Comment by icedrocket** [comment]
I just included it because I've seen other crates include it in the prelude module. I'll edit the commit.

**Comment by icedrocket** [comment]
I just included it because I've seen other crates include it in the prelude module. I'll edit the commit.

**Comment by icedrocket** [comment]
@jessebraham It seems that the `system` module is not exposed to users in `esp32-hal`, `esp32s2-hal` and `esp32s3-hal`. Is this intentional?

**Comment by icedrocket** [comment]
@jessebraham It seems that the `system` module is not exposed to the user in `esp32-hal`, `esp32s2-hal` and `esp32s3-hal`. Is this intended?

**Comment by icedrocket** [comment]
The reason I created this PR is because I discovered that the traits exposed by the prelude module are slightly inconsistent across platforms. For example, `SensExt` is exposed on esp32 and esp32s2 but not on esp32s3. Also, `SarAdcExt` is not exposed on esp32c2 and esp32c3. I guess this was not intended by the developers, so I fixed it and included most of the traits to make it easier to use them.

If there are traits that shouldn't be included in the prelude module, please let me know about those traits or feel free to add a new commit to remove them.

File: esp-hal-common/src/prelude.rs [hunk 1]
```diff
@@ -55,7 +98,50 @@ pub mod eh1 {
     };
     pub use nb;
 
-    pub use crate::system::SystemExt;
+    #[cfg(any(esp32c2, esp32c3))]
+    pub use crate::analog::SarAdcExt as _esp_hal_analog_SarAdcExt;
+    #[cfg(any(esp32, esp32s2, esp32s3))]
+    pub use crate::analog::SensExt as _esp_hal_analog_SensExt;
+    #[cfg(not(esp32c2))]
```
Comment (jessebraham) [inline esp-hal-common/src/prelude.rs:105]:
```suggestion
    #[cfg(rmt)]
```
Same as above

File: esp-hal-common/src/prelude.rs [hunk 2]
```diff
@@ -55,7 +98,50 @@ pub mod eh1 {
...
+    pub use crate::pulse_control::{
+        ConfiguredChannel as _esp_hal_pulse_control_ConfiguredChannel,
+        OutputChannel as _esp_hal_pulse_control_OutputChannel,
+    };
+    #[cfg(any(esp32, esp32s2))]
+    pub use crate::spi::dma::WithDmaSpi3 as _esp_hal_spi_dma_WithDmaSpi3;
+    pub use crate::{
+        clock::Clock,
```
Comment (jessebraham) [inline esp-hal-common/src/prelude.rs:113]:
```suggestion
        clock::Clock as _esp_hal_clock_Clock,
```
Same as above

**Review by jessebraham — CHANGES_REQUESTED** [review]
Sorry for the delay in getting to this, thanks for the contribution. Looks good overall, just a few small changes I'd like to see please!

**Pull request merged**